book/src/intro.md

```diff
@@ -284,28 +283,33 @@ SELECT avg(radius) as R, avg(width) as W FROM (
 {R:1.75,W:1.5}
 ```
 
-Things get more interesting when you want to do different types of processing
+Things get more interesting when you want to do different kinds of processing
 for differently typed entities, e.g., let's compute an average radius of circles,
 and double the width of each rectangle.  This time we'll use the pipe syntax
 with shortcuts and employ first-class errors to flag unknown types:
-```
+```sh
 $ super -c "
 values
   {kind:'circle',radius:1.5},
   {kind:'rect',width:2.0,height:1.0},
   {kind:'circle',radius:2},
-  {kind:'rect',width:1.0,height:3.5}
+  {kind:'rect',width:1.0,height:3.5},
+  {kind:'triangle',base:2.0,height:3.0}
 | switch kind
     case 'circle' (
         R:=avg(radius)
     )
     case 'rect' (
         width:=width*2
     )
+    default (
+        error({message:'unknown shape',value:this})
+    )
```

Comment:
I added this along with the "triangle" record above in attempt to make good on the "...and employ first-class errors to flag unknown types" statement that was already present in the text. Or if the original/shorter example is preferred, we could drop that part of the text.